### ISSUE TITLE
F37 & G13 missing info / outdated ?

### ISSUE THREAD

**Issue by goetsu** [w3c/wcag#304]
Hi,

I know that you are all busy working on WCAG 2.1 but I have a comments regarding https://www.w3.org/TR/2008/NOTE-WCAG20-TECHS-20081211/F37
and 
https://www.w3.org/TR/2008/NOTE-WCAG20-TECHS-20081211/G13
specially for the onchange on a select to redirect / open a new page case

I'm wondering if F37 must be updated because nowdays on windows browser (at least FF, Chrome, IE11) allow to open the list of a select using space key and then navigate the options using arrow key without triggering the onchange event.

Also regarding G13 it's not clear if the informative text may or not be visually hidden or added using a title for example. I know that it's not an ideal solution but asking to add a submit button for the kind of feature is a lost cause.

**Comment by patrickhlauke** [comment]
> I'm wondering if F37 must be updated because nowdays on windows browser (at least FF, Chrome, IE11) allow to open the list of a select using space key and then navigate the options using arrow key without triggering the onchange event.

it may not trigger while going up/down the list, but once the user moves on to the next form control `onchange` is still triggered, and it's still non-standard and confusing to the user who wasn't expecting it.

> Also regarding G13 it's not clear if the informative text may or not be visually hidden or added using a title for example.

visually hidden text would help AT users, but still leave the problem there for sighted/partially sighted users (potentially with cognitive disabilities or other factors at play).

**Comment by awkawk** [comment]
I think that https://www.w3.org/TR/WCAG20-TECHS/SCR19.html may also be instructive - there is no prohibition on using onchange in WCAG 2.0, unless you are causing a change of context. In modern browsers, and in fact in many older ones, the keyboard access makes it possible for users to fully navigate the options within a select (in older browsers it was possible but browsers are now making it more user friendly).

If the user agents you are relying on for your conformance claim don't provide a way for users to move through the options in a select without firing the onchange event and then causing a change of context, you won't pass 3.2.2, and if the keyboard support is there but the page jumps the user to a new page or some unexpected control when the change event is fired you also won't pass 3.2.2. SCR19 show one example of how to use onchange appropriately, although this still depends on the user agent supporting the keyboard access appropriately.

**Comment by goetsu** [comment]
@patrickhlauke the blur issue is something that can be fixed with js and currently the lack of visible information isn't a failure if you read the current test procedures

@awkawk in case of a select used for redirection / opening a new page I don't consider it as a change of context if user need to activate that change of context with enter key because it's an "explicit" decision of the user like activating a submit button or link

**Comment by awkawk** [comment]
@goetsu what you are describing is F37. From the Understanding 3.2.2 doc: "So checking a checkbox, entering text into a text field, or changing the selected option in a list control changes its setting, but activating a link or a button does not. "

**Comment by goetsu** [comment]
@awkawk yes I consider it a change of context if it trigger the redirection/open the new page when I use arrow key to change the selected option in the select or on blur of the select but not if user has to activate his choice using enter key because as I say I consider it's like activating a link or a button in that case

**Comment by awkawk** [comment]
@goetsu you can activate a button with the space bar and a checkbox with the space bar, is that the same thing also? :)

**Comment by fstrr** [comment]
Closing as marked as inactive in 2019. If this needs to be re-opened, please do so.
